enhancements/installer/ibmcloud-ipi.md

```diff
@@ -0,0 +1,181 @@
+
+
+### Non-Goals
+
+None at this time.
+
+## Proposal
+
+- Implement installer CLI prompts in order to build a default/minimal `install-config.yaml` for IBM Cloud :
+  ```shell
+  ? SSH Public Key /Users/someone/.ssh/id_rsa.pub
+  ? Platform ibmcloud
+  ? Resource Group ID default (34ffb674f7c4466398dcd257a0dac58e)
+  ? Region us-south
+  ? RHCOS Custom Image rhcos-ibmcloud-470
+  ? Base Domain ibm.foo.com (Internet Services-9h)
+  ? Cluster Name test
+  ? Pull Secret [? for help] ****
+  ```
+
+- Implement IBM Cloud specific platform and machine pool installer customizations
+  
+  **Platform**
+  
+  - region (required string): The IBM Cloud region where the cluster will be created.
+  - cisInstanceCRN (required string): The Cloud Internet Services CRN managing the base domain DNS zone.
+  - clusterOSImage (required string): The name of the RHCOS custom image to use for machines.
+  - resourceGroup (optional string): The name of an existing resource group where the cluster and all required resources will be created.
+  - defaultMachinePlatform (optional object): Default machine pool properties that apply to machine pools that do not define their own IBM Cloud specific properties.
+  
+  If one of these is specified, they ALL must be specified.
+  - vpc (optional string): The name of an existing VPC network.
+  - vpcResourceGroup (optional string): The name of the existing VPC's resource group.
+  - subnets (optional array of strings): A list of existing subnet IDs. Leave unset and the installer will create new subnets in the VPC network on your behalf.
+  
+  **Machine Pool**
+  - type (optional string): The VSI machine profile.
+  - zones (optional array of strings): The availability zones used for machines in the pool.
+  - bootVolume (optional object):
+    - encryptionKeyCRN: (optional string): The CRN referencing a a Key Protect or Hyper Protect Crypto Services key to use for volume encryption. If not specified, a provider managed encryption key will be used.
+
+- Provide documentation (in similar format to other supported IPI platforms) to help users use and customize the IPI installer on IBM Cloud:
+  - Prerequisite instructions prior to invoking installer
+  - Description of installer options and customizations as they apply to IBM Cloud
+  - Post installation instructions on further customizations a user may wish to apply
+
+### User Stories
+
+Story 1
+As an OpenShift consumer, I want to quickly, with minimal input and default options, be able to use the OpenShift installer to create and destroy an OpenShift 4 cluster on IBM Cloud.
+
+Story 2
+As an OpenShift consumer, I want to be able to use the OpenShift installer with customizations (i.e. enhanced security) to create and destroy an OpenShift 4 cluster on IBM Cloud.
+
+Story 3
+As an OpenShift platform, I want OpenShift on IBM Cloud to be maintained and released like other supported OpenShift platforms and covered by CI tests.
```

Comment:
This may be a step too far in anthropomorphizing. Who actually wants this? The OpenShift consumer, maybe?